sklearn/utils/testing.py

```diff
@@ -647,8 +648,20 @@ def is_abstract(c):
     return sorted(set(estimators))
 
 
+NON_RANDOM_CLASSES = (DBSCAN,)
+
+
 def set_random_state(estimator, random_state=0):
```

Comment:
I am wondering if it is OK to do it here. The other alternative is to set `random_state` to `None` is `get_optimal_parameters` and ensure it is always called after `set_random_state`